src/modding/WorkshopData.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using Godot;
+using Newtonsoft.Json;
+using File = Godot.File;
+
+/// <summary>
+///   Holds data that we need to interact with the Steam workshop
+/// </summary>
+public class WorkshopData
+{
+    [JsonProperty]
+    public Dictionary<string, ulong> KnownModWorkshopIds { get; private set; } = new();
+
+    public static WorkshopData Load()
+    {
+        using var file = new File();
+        if (!file.FileExists(Constants.WORKSHOP_DATA_FILE))
+            return new WorkshopData();
+
+        if (file.Open(Constants.WORKSHOP_DATA_FILE, File.ModeFlags.Read) != Error.Ok)
+            throw new IOException("Can't read workshop data file even though it exists");
+
+        var data = file.GetAsText();
+
+        return JsonSerializer.Create().Deserialize<WorkshopData>(new JsonTextReader(new StringReader(data)));
```

Comment:
Why not just `return JsonConvert.DeserializeObject<WorkshopData>(data);`?

Reply:
Unless my ide was wrong there is no direct string taking method here (or maybe it was the writing from where I copy pasted and changed things enough to make it work) so I needed to create the extra objects.

Reply:
I see now you talked about using JsonConvert, which is actually a specific type to JSON converter, which are used in the game to only convert single instances and not do the entire to JSON operation, so I think that's good enough reason to not use that here.

~~I think that's related to the other convert C# operations? I don't use that here as I explicitly want to serialize as JSON not just do a bit unclearer conversion to JSON operation.~~